File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/bids/dto/BidResultResponse.java
package com.portfolio.auctionmarket.domain.bids.dto;

import com.portfolio.auctionmarket.domain.bids.entity.Bid;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BidResultResponse {

    private Long auctionId;
    private String title;
    private Long price;
    private Long currentPoint;

    public static BidResultResponse from(Bid entity) {
        return BidResultResponse.builder()
                .auctionId(entity.getAuction().getAuctionId())
                .title(entity.getAuction().getProduct().getTitle())
                .price(entity.getBidPrice())
                .currentPoint(entity.getBidder().getPoint())
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/bid/dto/BidResultResponse.kt
package org.example.auctioncommon.domain.bid.dto

import org.example.auctioncommon.domain.bid.entity.Bid

data class BidResultResponse(
    val auctionId: Long,
    val title: String,
    val price: Long,
    val currentPoint: Long,
)

fun Bid.toDto(): BidResultResponse {
    return BidResultResponse(
        auctionId = this.auction?.auctionId ?: 0L,
        title = this.auction?.product?.title ?: "제목이 없습니다.",
        price = this.bidPrice,
        currentPoint = this.bidder?.point ?: 0L
    )
}